clippy_utils/src/lib.rs

```diff
@@ -1473,6 +1473,16 @@ pub fn is_self_ty(slf: &hir::Ty<'_>) -> bool {
     false
 }
 
+/// Checks if a given type looks safe to be uninitialized.
+pub fn is_uninit_ty_valid(cx: &LateContext<'_>, ty: Ty<'_>) -> bool {
+    match ty.kind() {
+        rustc_ty::Array(component, _) => is_uninit_ty_valid(cx, component),
+        rustc_ty::Tuple(types) => types.types().all(|ty| is_uninit_ty_valid(cx, ty)),
+        rustc_ty::Adt(adt, _) => match_def_path(cx, adt.did, &paths::MEM_MAYBEUNINIT),
```

Comment:
Could you update this case to use lang_items instead of the paths to check for the type? The code would look something like (I haven't tested this yet, but it should work):
```suggestion
        rustc_ty::Adt(adt, _) => {
            cx.tcx.lang_items().maybe_uninit() == Some(adt.did)
        },
```